src/vs/workbench/api/browser/mainThreadTerminalService.ts

```diff
@@ -221,11 +222,18 @@ export class MainThreadTerminalService implements MainThreadTerminalServiceShape
 		// Proxy profile provider requests through the extension host
 		this._profileProviders.set(id, this._terminalService.registerTerminalProfileProvider(extensionIdentifier, id, {
 			createContributedTerminalProfile: async (options) => {
-				return this._proxy.$createContributedProfileTerminal(id, options);
+				return this._proxy.$createContributedProfileTerminal(id, this._resolveOptions(options));
 			}
 		}));
 	}
 
+	private _resolveOptions(options: any): ICreateContributedTerminalProfileOptions {
+		if (options.location && typeof options.location === 'object' && 'parentTerminal' in options.location) {
+			options.location = { splitActive: true };
+		}
```

Comment:
Let's convert to TerminalIdentifier instead so it's definitely pointing at that terminal